src/server/api.php

```diff
@@ -1009,17 +1024,69 @@ function meta_delphi() {
       }
     }
   } else if($source === 'sensors') {
-    if(require_all($data, array('auth', 'names', 'locations', 'epiweeks'))) {
-      if($_REQUEST['auth'] === $AUTH['sensors']) {
-        // parse the request
-        $names = extract_values($_REQUEST['names'], 'str');
-        $locations = extract_values($_REQUEST['locations'], 'str');
-        $epiweeks = extract_values($_REQUEST['epiweeks'], 'int');
-        // get the data
-        $epidata = get_sensors($names, $locations, $epiweeks);
-        store_result($data, $epidata);
+    if(require_all($data, array('names', 'locations', 'epiweeks'))) {
+      if(!array_key_exists('auth', $_REQUEST)) {
+        $auth_tokens_presented = array();
       } else {
-        $data['message'] = 'unauthenticated';
+        $auth_tokens_presented = extract_values($_REQUEST['auth'], 'str');
+      }
+      $names = extract_values($_REQUEST['names'], 'str');
+      $n_names = count($names);
+      $n_auth_tokens_presented = count($auth_tokens_presented);
+      $max_valid_granular_tokens_per_name = max(array_map("count", $GRANULAR_SENSOR_AUTH_TOKENS));
```

Comment:
Lack of PHP knowledge on my part here. The [docs](https://www.php.net/manual/en/function.array-map.php) say `array_map` takes a function and an array; but this is a string and an array. Does it know you want to use the `count` function?

Reply:
This is uncertain area for me too.  From some examples, it looks like you pass "functionname" for built-ins and normally declared functions, and $functionvar if you have a function pointer.  I need to check on the name resolution rules for the string input.

Reply:
Ok, good to know, thanks.